src/main/java/io/logz/jmx2logzio/listener/prometheus/PromListenerWriter.kt

```diff
@@ -0,0 +1,123 @@
+package io.logz.jmx2logzio.listener.prometheus
+
+import io.logz.jmx2logzio.Utils.HangupInterceptor
+import io.logz.jmx2logzio.listener.ListenerWriter
+import io.logz.jmx2logzio.objects.LogzioJavaSenderParams
+import io.logz.jmx2logzio.objects.Metric
+import io.logz.jmx2logzio.objects.StatusReporterFactory
+import io.logz.prometheus.protobuf.Types
+import io.logz.prometheus.protobuf.Types.Sample
+import io.logz.prometheus.protobuf.Types.TimeSeries
+import io.logz.sender.HttpsRequestConfiguration
+import io.logz.sender.exceptions.LogzioParameterErrorException
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import javax.annotation.PostConstruct
+
+class PromListenerWriter(
+    senderParams: LogzioJavaSenderParams
+) : ListenerWriter {
+    companion object {
+        private val logger = LoggerFactory.getLogger(PromListenerWriter::class.java)
+    }
+
+    private val senderExecutors: ScheduledExecutorService = Executors.newScheduledThreadPool(senderParams.threadPoolSize);
+    private val sender: PromLogzioSender = createSender(senderExecutors, senderParams)!!.also {
+        it.start()
+    }
+
+    private fun createSender(senderExecutors: ScheduledExecutorService, logzioSenderParams: LogzioJavaSenderParams): PromLogzioSender? {
+        try {
+            val statusReporter = StatusReporterFactory.newSenderStatusReporter(LoggerFactory.getLogger(logzioSenderParams.loggerName))
+
+            val requestConf = HttpsRequestConfiguration
+                .builder()
+                .setLogzioListenerUrl(logzioSenderParams.url)
+                .setLogzioType(logzioSenderParams.type)
```

Comment:
Type?

Reply:
It looked like a mandatory param, but I will try without it